docs/examples/sentence_classification.md

```diff
@@ -0,0 +1,344 @@
+
+The confusion matrix:
+
+
+```python
+cm
+```
+
+
+
+
+               False    True
+       False    4624     201
+        True      41     706
+
+
+
+
+```python
+model.draw()
+```
+
+
+
+
+    
+![svg](sentence_classification_files/sentence_classification_14_0.svg)
+    
+
+
+
+Now let's try to use logistic regression to classify messages. I will use different tips to make my model perform better. As in the previous example, I artificially increase the number of spam. The logistics regression will be fed from a tf-idf.
+
+
+```python
+from river import linear_model
+from river import optim
+from river import preprocessing
+
+X_y = datasets.SMSSpam()
+
+model = extract_body | feature_extraction.TFIDF() | preprocessing.StandardScaler() | imblearn.RandomUnderSampler(
+    classifier=linear_model.LogisticRegression(optimizer=optim.SGD(0.05), loss=optim.losses.Log()),
+    desired_dist={0: 0.5, 1: 0.5},
+    seed=42
+)
+
+metric = metrics.ROCAUC()
+
+cm = metrics.ConfusionMatrix()
+
+for x, y in X_y:
+    
+    y_pred = model.predict_one(x)
+
+    metric.update(y_pred=y_pred, y_true=y)
+    
+    cm.update(y_pred=y_pred, y_true=y)
+    
+    model.learn_one(x, y)
+    
+metric
+```
+
+
+
+
+    ROCAUC: 0.834134
+
+
+
+The confusion matrix:
+
+
+```python
+cm
+```
+
+
+
+
+               False    True
+       False    4027     800
+        True     124     623
+
+
+
+
+```python
+model.draw()
+```
+
+
+
+
+    
+![svg](sentence_classification_files/sentence_classification_19_0.svg)
+    
+
+
+
+The results of the logistic regression are less good. It is more difficult for the model to determine if a text message is spam.
+
+Let's try to use word embeddings to improve our logistic regression. Word embeddings allow you to represent a word as a vector. Embeddings are developed to build semantic rich vectors, i.e. the vector which represents the word **python** is close to the vector which represents the word **programming**. We will use Scipy to convert our sentence to vectors. Scipy convert a sentence to a vector by calculating the average of the embeddings of the words of the sentence.
+
+You can download pre-trained embeddings in many languages. We will use English pre-trained embeddings as our sms are in English.
+
+The command below allows you to download the embeddings of spacy. More informations about spacy and it's installation [here](https://spacy.io/usage).
+python -m spacy download en_core_web_sm
+Here, I create a custom transformer to convert an input sentence to a dict of float. I will integrate this transformer into my pipeline.
+
+
+```python
+import spacy
+
+from river.base import Transformer
+
+class Embeddings(Transformer):
+    """My custom transformer, words embeddings using spacy."""
+    
+    def __init__(self):
+        self.embeddings = spacy.load('en_core_web_sm')
+        
+    def transform_one(self, x, y=None):
+        return {i: xi for i, xi in enumerate(self.embeddings(x).vector)}
```

Comment:
I would give more meaning to `i`, for instance call it `dimension` or something like that.